kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -89,31 +90,44 @@ public KafkaProxy startup() throws InterruptedException {
             throw new IllegalStateException("This proxy is already running");
         }
 
+        var portConflictDefector = new PortConflictDetector();
+        portConflictDefector.validate(virtualClusterMap);
+
         var availableCores = Runtime.getRuntime().availableProcessors();
         var meterRegistries = new MeterRegistries(micrometerConfig);
 
-        this.adminEventGroup = buildNettyEventGroups(availableCores, config.isUseIoUring());
-        this.serverEventGroup = buildNettyEventGroups(availableCores, config.isUseIoUring());
+        this.adminEventGroup = buildNettyEventGroups("admin", availableCores, config.isUseIoUring());
+        this.serverEventGroup = buildNettyEventGroups("server", availableCores, config.isUseIoUring());
 
         maybeStartMetricsListener(adminEventGroup, meterRegistries);
 
-        this.endpointProviders = virtualClusterMap.entrySet().stream()
-                .collect(Collectors.toMap(e -> e.getValue().getClusterEndpointProvider(), Map.Entry::getValue));
+        var tlsServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, true, endpointRegistry, false, Map.of()));
+        var plainServerBootstrap = buildServerBootstrap(serverEventGroup, new KafkaProxyInitializer(config, false, endpointRegistry, false, Map.of()));
 
-        var networkBindings = buildNetworkBindings();
+        var unused = networkBindingExecutor.submit(() -> {
+            try {
+                do {
```

Comment:
I'm thinking a safety try/catch in the loop would be nice, even though the methods it's calling now look safe it would be painful if a change caused some unexpected error that quietly kills the loop without logging out a stacktrace somewhere.